woocommerce/woocommerce
# Fix: `useEditorBlocks` hook always returns an array

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



`useEntityBlockEditor` can return undefined. This causes the crash of the application because an array is expected. I created a fallback logic.





Closes #41873  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you have already generated a website with the AI.
2. Head over to Home > Customize your Store.
3. Click on Customize.
4. Ensure that you don't see a popup with this error: `The editor has encountered an unexpected error`.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Fix: `useEditorBlocks` hook always returns an array.


#### Comment 

</details>
